swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -689,6 +691,19 @@ predicate storeStep(Node node1, ContentSet c, Node node2) {
     node2 = node1 and // TODO: again, we should ideally have a separate Node case here, and not reuse the CallExpr
     c instanceof OptionalSomeContentSet and
     init.isFailable()
+  ) or
+  // creation of an array `[v1,v2]`
+  exists(ArrayExpr arr |
+    node1.asExpr() = arr.getAnElement() and
+    node2.asExpr() = arr and
+    c.isSingleton(any(Content::ArrayContent ac))
+  ) or
+  // array assignment `a[n] = x`
+  exists(AssignExpr assign, SubscriptExpr subscript |
+    node1.asExpr() = assign.getSource() and
+    node2.(PostUpdateNode).getPreUpdateNode().asExpr() = subscript.getBase() and
+    subscript = assign.getDest() and
+    subscript.getBase().getType().(InOutType).getObjectType() instanceof ArrayType
```

Comment:
Is this last condition necessary? Isn't a write to a subscript expression enough to conclude that we should have a `storeStep`?

Reply:
Yes - `SubscriptExpr` can be used with dictionaries, among other types.

Reply:
Ah, I see. Fair enough, then 👍